bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1046,6 +1047,8 @@ void ReplicaImp::onMessage<FullCommitProofMsg>(FullCommitProofMsg *msg) {
       const bool askForMissingInfoAboutCommittedItems =
           (msgSeqNum > lastExecutedSeqNum + config_.getconcurrencyLevel());  // TODO(GG): check/improve this logic
 
+      consensus_times_.end(msgSeqNum);
```

Comment:
Isn't it better to put this call at the beginning of executeRequestsAndSendResponses() function to have just one closer point?

Reply:
It's a good point. Then we only need to call it once also.

Reply:
We don't actually have the sequence number in that call so we can't do this without modifying the API, which I don't want to do in this commit.

My original thinking was also that we only want to measure consensus. Consensus is complete before that call. It's probably not a big difference, but we don't even guarantee that anything gets executed at that point as there can be gaps in the log or a missing preprepare. To measure the time between consensus and execution we will need at least one more histogram. That work can be done in future commits.

Reply:
I ended up changing the API to take the seqnumber and doing it there as you suggested. It is indeed cleaner.